core/src/main/scala/org/apache/spark/deploy/StandaloneSparkHadoopUtil.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy
+
+import java.security.PrivilegedAction
+
+import org.apache.hadoop.mapred.JobConf
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.spark.{SparkEnv, Logging}
+import sun.security.jgss.krb5.Krb5InitCredential
+
+import scala.sys.process.Process
+
+private[spark] class StandaloneSparkHadoopUtil extends SparkHadoopUtil {
+
+  val hadoopSecurityAuthenticationKey = "hadoop.security.authentication"
```

Comment:
I was debating that - would like @pwendell to weigh in here. I figured it would be better to reduce information redundancy; presumably hdfs-site.xml configurations already defined these principals and keytabs. But I'm probably just splitting hairs at that point.